app/db/interface.py

```diff
@@ -139,12 +140,85 @@ def _remove_soft_deletes(self, payments):
         """
         Takes a list of payments and removes those payments that have been 'soft deleted,'
         meaning the 'is_removed' field has been set to True
-        
+
         Maybe there's a probably a better way to do this by including requirement in db query,
-        but this will do for now 
+        but this will do for now
         """
         return [payment for payment in payments if not payment.is_removed]
-        
+
+    def perform_payment_action(self, user_id, payment_attributes=None):
+        """
+        Accepts a payment with user_id and performs an action on it
+
+        :param user_id: <int> the unique identifier of a user_id whose payment/s is/are to be actioned upon
+        :param payment_attributes: <dict> a collection of new payment attribute values that will update old ones
+
+        if payment_attributes has payment_id, then it is "set-default" action.
+        This also sets any other default payment of the same user to False
+
+        if payment_attributes has amount, then it is performing "charge" action.
+        This would update the charge history
+        """
+        try:
+            payments = self._query_payments(payment_attributes={"user_id":user_id})
+            if 'payment_id' in payment_attributes: #for set-default action
+                payment_id_to_be_updated = payment_attributes['payment_id']
+                payment_to_be_updated = None
+                for payment in payments:
+                    if payment.id == payment_id_to_be_updated: #no need to check if is_removed = False because the query filters it
+                        payment_to_be_updated = payment
+                if payment_to_be_updated != None:
```

Comment:
It's slightly a matter of style, but to be safe it is best to use `is not` with `None`.